libraries/AP_Math/AP_Math.cpp

```diff
@@ -1,11 +1,22 @@
 #include "AP_Math.h"
 #include <float.h>
 
-template <class FloatOne, class FloatTwo>
-bool is_equal(const FloatOne v_1, const FloatTwo v_2)
+template <class Arithmetic1, class Arithmetic2>
+typename std::enable_if<std::is_integral<typename std::common_type<Arithmetic1, Arithmetic2>::type>::value ,bool>::type
+is_equal(const Arithmetic1 v_1, const Arithmetic2 v_2)
 {
-    static_assert(std::is_arithmetic<FloatOne>::value, "template parameter not of type float or int");
-    static_assert(std::is_arithmetic<FloatTwo>::value, "template parameter not of type float or int");
+    static_assert(std::is_arithmetic<Arithmetic1>::value, "template parameter not of type int");
+    static_assert(std::is_arithmetic<Arithmetic2>::value, "template parameter not of type int");
```

Comment:
Do you really need the static asserts? The function shouldn't even be generated for other types.